pkg/autoupdate/handler.go

```diff
@@ -81,6 +82,24 @@ func (u *Updater) handler() tuf.NotificationHandler {
 			return
 		}
 
+		// Check that it all came through okay
+		if err := checkExecutable(context.TODO(), outputBinary, "--version"); err != nil {
+			level.Error(u.logger).Log(
+				"msg", "Broken updated binary. Removing",
+				"target", u.target,
+				"outputBinary", outputBinary,
+				"err", err,
+			)
+			if err := os.RemoveAll(updateDir); err != nil {
+				level.Error(u.logger).Log(
```

Comment:
Just brainstorming here.. Suppose we _do_ fail to remove the corrupt update, the next time launcher starts up we'll just run through the same process. Continually failing to remove the corrupt update won't prevent the launcher binary from updating, so it probably doesn't make sense to try other mitigation methods (I was initially thinking of re-downloading the latest update and putting it a 'newer' directory or something..).

In any case, we'll (hopefully) be able to look at the logs on the device if we need to investigate further.

kinda makes me wish we had a canary style test, e.g. a launcher that pulls updates and fails very loudly somehow if the update is corrupt. Though, the corruption problem is probably _not_ related to us serving corrupt updates..

Reply:
(that got long and rambly...)

Reply:
Yeah... I don't really know the history of the current update model, but I kinda want to majorly re-write it. Right now, it has an asynchronous thread which downloads an update, and calls a `handler` to trigger a restart. 

This design makes it hard to have clear control over how things are happening. Or even to trigger a `check now` style function. It's all very `fire and forgot` style.

But I agree -- this error shouldn't end up being fatal. It means there's some weird detritus on disk.